elasticdl/python/master/servicer.py

```diff
@@ -414,16 +410,23 @@ def _process_gradients(
                 else:
                     self._gradient_sum[k] = v
             self._grad_n += 1
-        else:
-            # TODO: do not accumulate gradients but apply directly.
-            pass
 
         # staleness-aware learning rate modulation
         if self._lr_modulation:
             staleness = max(1, self._version - request_version)
             self._lr_modulation.set_multiplier(1.0 / staleness)
         if self._use_async or self._grad_n >= self._grad_to_wait:
-            self._update_model()
+            if not self._use_async:
```

Comment:
The logic is a bit hard to read here since we have both:
```
if self._use_async or self._grad_n >= self._grad_to_wait:
```
and then 
```
if not self._use_async:
```
Even though it is valid by taking a closer look but it would be great if there's a good way to re-arrange the if conditions here.

Reply:
Revised.